app/code/Magento/QuoteGraphQl/Model/Resolver/Coupon/ApplyCouponToCart.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\QuoteGraphQl\Model\Resolver\Coupon;
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Exception\GraphQlNoSuchEntityException;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\Resolver\ValueFactory;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Quote\Api\CouponManagementInterface;
+use Magento\Quote\Model\MaskedQuoteIdToQuoteIdInterface;
+use Magento\QuoteGraphQl\Model\CartMutationsAllowedInterface;
+
+/**
+ * {@inheritdoc}
+ */
+class ApplyCouponToCart implements ResolverInterface
+{
+    /**
+     * @var CouponManagementInterface
+     */
+    private $couponManagement;
+
+    /**
+     * @var ValueFactory
+     */
+    private $valueFactory;
+
+    /**
+     * @var MaskedQuoteIdToQuoteIdInterface
+     */
+    private $maskedQuoteIdToQuoteId;
+
+    /**
+     * @var CartMutationsAllowedInterface
+     */
+    private $cartMutationsAllowed;
+
+    /**
+     * @param ValueFactory $valueFactory
+     * @param CouponManagementInterface $couponManagement
+     * @param MaskedQuoteIdToQuoteIdInterface $maskedQuoteIdToId
+     * @param CartMutationsAllowedInterface $cartMutationsAllowed
+     */
+    public function __construct(
+        ValueFactory $valueFactory,
+        CouponManagementInterface $couponManagement,
+        MaskedQuoteIdToQuoteIdInterface $maskedQuoteIdToId,
+        CartMutationsAllowedInterface $cartMutationsAllowed
+    ) {
+        $this->valueFactory = $valueFactory;
+        $this->couponManagement = $couponManagement;
+        $this->maskedQuoteIdToQuoteId = $maskedQuoteIdToId;
+        $this->cartMutationsAllowed = $cartMutationsAllowed;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null) : Value
+    {
+        $maskedQuoteId = $args['input']['cart_id'];
+        $couponCode = $args['input']['coupon_code'];
+
+        if (!$maskedQuoteId || !$couponCode) {
+            throw new GraphQlInputException(__('Required parameter is missing'));
```

Comment:
It is better to specify which parameter is missing (have two separate exceptions)